tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java

```diff
@@ -1136,6 +1142,41 @@ private boolean deleteDagDirectories(Channel channel,
       return false;
     }
 
+    private boolean deleteTaskAttemptDirectories(Channel channel, List<String> taskAttemptFailedQ,
+                                            List<String> jobQ, List<String> dagIdQ, List<String> taskAttemptIdQ) {
+      if (jobQ == null || jobQ.isEmpty()) {
+        return false;
+      }
+      if (taskAttemptFailedQ != null && !taskAttemptFailedQ.isEmpty() && taskAttemptFailedQ.get(0).contains("delete")
```

Comment:
I can see this boilerplate in more places in shufflehandler, which is because of the nature of QueryStringDecoder.get(...) -> List<String>
could you please refactor this into common methods, like:
```
notEmpty(taskAttemptFailedQ)
notEmptyAndContains(taskAttemptFailedQ, "delete")
```

Reply:
ack